coldfront/plugins/qumulo/views/update_allocation_view.py

```diff
@@ -90,38 +147,47 @@ def form_valid(
             end_date_extension=10,
         )
 
-        # NOTE - "storage_protocols" will have special handling
-        attributes_to_check = [
-            "cost_center",
-            "department_number",
-            "technical_contact",
-            "billing_contact",
-            "service_rate",
-            "storage_ticket",
-            "storage_quota",
-        ]
+        storage_quota_attribute = AllocationAttribute.objects.get(
+            allocation_attribute_type=AllocationAttributeType.objects.get(
+                name="storage_quota"
+            ),
+            allocation=allocation,
+        )
+
+        if storage_quota_attribute.value != form_data.get("storage_quota"):
+            AllocationAttributeChangeRequest.objects.create(
+                allocation_attribute=storage_quota_attribute,
+                allocation_change_request=allocation_change_request,
+                new_value=form_data.get("storage_quota"),
+            )
 
-        form_values = [form_data.get(field_name) for field_name in attributes_to_check]
+        storage_protocols = json.dumps(form_data.get("protocols"))
 
-        # handle "storage_protocols" separately
-        attributes_to_check.append("storage_protocols")
-        form_values.append(json.dumps(form_data.get("protocols")))
+        storage_protocols_attribute = AllocationAttribute.objects.get(
+            allocation_attribute_type=AllocationAttributeType.objects.get(
+                name="storage_protocols"
+            ),
+            allocation=allocation,
+        )
 
-        for attribute_name, form_value in zip(attributes_to_check, form_values):
-            UpdateAllocationView._handle_attribute_change(
-                allocation=allocation,
+        if storage_protocols_attribute.value != storage_protocols:
+            AllocationAttributeChangeRequest.objects.create(
+                allocation_attribute=storage_protocols_attribute,
                 allocation_change_request=allocation_change_request,
-                attribute_name=attribute_name,
-                form_value=form_value,
+                new_value=storage_protocols,
             )
 
-        # RW and RO users are not handled via an AllocationChangeRequest
         access_keys = ["rw", "ro"]
         for key in access_keys:
             access_users = form_data[key + "_users"]
             self.set_access_users(key, access_users, allocation)
 
-        return super().form_valid(form=form, parent_allocation=parent_allocation)
+    def form_valid(self, form: UpdateAllocationForm):
+        if 'reset_acls' in self.request.POST:
```

Comment:
@ris-bmulligan why is the method a POST? I thought it should be PUT or DELETE?

Reply:
The submit button does a POST.